ibis/backends/risingwave/__init__.py

```diff
@@ -268,3 +317,262 @@ def list_databases(
+        Table
+            Table expression
+        """
+        column_defs = [
+            sge.ColumnDef(
+                this=sg.to_identifier(colname, quoted=self.compiler.quoted),
+                kind=self.compiler.type_mapper.from_ibis(typ),
+                constraints=(
+                    None
+                    if typ.nullable
+                    else [sge.ColumnConstraint(kind=sge.NotNullColumnConstraint())]
+                ),
+            )
+            for colname, typ in schema.items()
+        ]
+
+        table = sg.table(name, db=database, quoted=self.compiler.quoted)
+        target = sge.Schema(this=table, expressions=column_defs)
+
+        create_stmt = sge.Create(
+            kind="SOURCE",
+            this=target,
+            properties=sge.Properties(
+                expressions=sge.Properties.from_dict(connector_properties)
+            ),
+        )
+
+        create_stmt = create_stmt.sql(self.dialect) + data_and_encode_format(
+            data_format, encode_format, encode_properties
+        )
+
+        with self._safe_raw_sql(create_stmt):
+            pass
+
+        return self.table(name, database=database)
+
+    def drop_source(
+        self,
+        name: str,
+        *,
+        database: str | None = None,
+        force: bool = False,
+    ) -> None:
+        """Drop a Source.
+
+        Parameters
+        ----------
+        name
+            Source name to drop.
+        database
+            Name of the database where the view exists, if not the default.
+        force
+            If `False`, an exception is raised if the source does not exist.
+        """
+        src = sge.Drop(
+            this=sg.table(name, db=database, quoted=self.compiler.quoted),
+            kind="SOURCE",
+            exists=force,
+        )
+        with self._safe_raw_sql(src):
+            pass
+
+    def create_sink(
+        self,
+        name: str,
+        sink_from: str | None = None,
+        connector_properties: dict | None = None,
+        *,
+        obj: ir.Table | None = None,
+        database: str | None = None,
+        data_format: str | None = None,
+        encode_format: str | None = None,
+        encode_properties: dict | None = None,
+    ) -> None:
+        """Creating a sink.
+
+        Parameters
+        ----------
+        name
+            Sink name to Create.
+        sink_from
+            The table or materialized view name to sink from. Only one of `sink_from` or `obj` can be
+            provided.
+        connector_properties
+            The properties of the sink connector, providing the connector settings to push to the downstream data sink.
+            Refer https://docs.risingwave.com/docs/current/data-delivery/ for the required properties of different data sink.
+        obj
+            An Ibis table expression or pandas table that will be used to extract the schema and the data of the new table. Only one of `sink_from` or `obj` can be provided.
```

Comment:
this should probably be pandas DataFrame and the type hint above should also include `pd.DataFrame` -- however, is this supported / tested?

Reply:
Yes, you are right. `pandas table` should not be added. It should be a typo. Updated